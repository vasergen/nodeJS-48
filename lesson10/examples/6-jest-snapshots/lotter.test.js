const { lottery } = require("./lottery");

const mockGenerateNumber = jest.fn();
jest.mock("./generateNumber", () => {
  return {
    generateNumber: () => mockGenerateNumber(),
  };
});

// console.log("!!!mockGenerateNumber", mockGenerateNumber);

describe("lottery", () => {
  it("should win", () => {
    mockGenerateNumber.mockImplementation(() => 5);

    const result = lottery(5);
    expect(result).toMatchInlineSnapshot(`"You won"`);
  });

  it("should lost", () => {
    mockGenerateNumber.mockImplementation(() => 1);

    const result = lottery(5);
    expect(result).toMatchInlineSnapshot(`"You lost, generated number: 1!"`);
  });

  it("should lost, expected 5 got 2", () => {
    mockGenerateNumber.mockImplementation(() => 2);

    const result = lottery(5);
    expect(result).toMatchSnapshot();
  });
});
