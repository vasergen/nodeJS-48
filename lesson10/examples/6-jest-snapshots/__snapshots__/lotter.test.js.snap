// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lottery should lost, expected 5 got 2 1`] = `"You lost, generated number: 2!"`;
